src/main/webapp/app/home/home.component.ts

```diff
@@ -185,6 +185,8 @@ export class HomeComponent implements OnInit, AfterViewChecked {
             const redirect = this.stateStorageService.getUrl();
             if (redirect && redirect !== '') {
                 this.stateStorageService.storeUrl('');
+                // TODO: this does not work with query parameters in the URL, they somehow get escaped, e.g. https://artemis.ase.in.tum.de/courses/10/exercises/5017%3Fjwt%3Dabc
+                // instead of https://artemis.ase.in.tum.de/courses/10/exercises/5017?jwt=abc
                 this.router.navigate([redirect]);
```

Comment:
In theory, this should work:
```suggestion
                this.router.navigateByUrl([redirect]);
```

Reply:
thanks!